app/models/pbs/mailing_list.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2022, Pfadibewegung Schweiz. This file is part of
+#  hitobito_pbs and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_pbs.
+
+module Pbs::MailingList
+  extend ActiveSupport::Concern
+
+  included do
+    validate :assert_layer_suffix
+
+    private
+
+    def assert_layer_suffix
+      return unless mail_name_changed?
```

Comment:
Weshalb nur wenn der mail_name geändert wurde? So wie ich das verstanden habe, ist das Ziel langsam aber sicher alle Abos entsprechend anzupassen.
Dann macht es vielleicht auch Sinn die Validierung bei sämtlichen Changes auszulösen.
So oder so wäre ein CHANGELOG Eintrag aber sicher auch super, damit das für den Nutzer nicht aus dem Nichts kommt :)

Reply:
Diese neue Validierung macht viele (vermutlich die meisten) MailingLists in der DB invalid. Manchmal validiert Rails auch related Models, also z.B. wenn ich eine neue Message erstelle und diese validiere, dann könnte es sein dass Rails auch die MailingList validiert, und dann die nicht ganz einfach verständliche Message "Abo ist ungültig" anzeigt. Ich durchschaue nicht ganz, wann Rails das macht, aber es war schon einmal ein Problem, als wir bei der PBS eine neue Validierung auf der Gruppe eingeführt haben, und die Leute dann Anlässe in dieser Gruppe nicht mehr bearbeiten konnten. Wenn wir es also immer aktivieren müsste man auf Integration viel besser testen ob noch alles was mit Abos zu tun hat weiterhin funktioniert. @Michael-Schaer was meinst du?

Eintrag im Changelog mach ich gleich noch :+1:

Reply:
Ah an diese related Models Geschichte hab ich nicht gedacht. Dann macht das natürlich schon viel mehr Sinn 👍🏼